cloudygames/views.py

```diff
@@ -1,28 +1,104 @@
 from django.shortcuts import render
+from django.core import serializers
 
-from rest_framework import viewsets, generics, status
+from rest_framework import viewsets, generics, status, filters
 from rest_framework.decorators import api_view
 from rest_framework.response import Response
 
 from cloudygames.serializers import GameSerializer, GameSessionSerializer, PlayerSaveDataSerializer
 from cloudygames.models import Game, GameSession, PlayerSaveData
 
+import django_filters
+import json
+
+
+######################## Filter ############################
+
+
+class GameFilter(django_filters.FilterSet):
+    users = django_filters.CharFilter(name='users__username')
+
+    class Meta:
+        model = Game
+        fields = ['id', 'name', 'publisher', 'users']
+        order_by = ['name']
+        read_only_fields = ('id',)
+
+class GameSessionFilter(django_filters.FilterSet):
+    game = django_filters.CharFilter(name='game__id')
+    player = django_filters.CharFilter(name='player_username')
+
+    class Meta:
+        model = GameSession
+        fields = ['game', 'player']
+
+class PlayerSaveDataFilter(django_filters.FilterSet):
+    game = django_filters.CharFilter(name='game__id')
+    player = django_filters.CharFilter(name='player_username')
+
+    class Meta:
+        model = PlayerSaveData
+        fields = ['game', 'player']
+
+
+######################## ViewSet ############################
+
 class GameViewSet(viewsets.ModelViewSet):
     serializer_class = GameSerializer
+    filter_class = GameFilter
 
     def get_queryset(self):
         is_owned = self.request.query_params.get('owned', 0)
-        if is_owned:
+        if is_owned == '1':
             _user = self.request.user
-            queryset = Game.objects.filter(users=_user)
-        else:
-            queryset = Game.objects.all()
-        return queryset.order_by('name')
+            return Game.objects.filter(users=_user)
+        return Game.objects.all().order_by('name')
 
 class GameSessionViewSet(viewsets.ModelViewSet):
```

Comment:
GameSession.player fields in the serializer must be automatically binded upon creation with a particular user's token as its player. If the user didn't pass a token, you can assume that player field is empty.

That means regardless what the player value is the API should ignore player field received and use the token instead.